### PR TITLE
Improve documentation of throwing methods [11820]

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
Both doxygen and user documentation is not complete lacking thorough information about the methods that can throw.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
Windows CI failure is unrelated to this PR that only modifies documentation.

File: src/cpp/database/database.hpp [hunk 1]
```diff
@@ -267,13 +274,15 @@ class Database
             const std::string& name) const;
 
     /**
-     * Get the entity of a given EntityKind that matches with the requested GUID
+     * @brief Get the entity of a given EntityKind that matches with the requested GUID.
      *
-     * If the given EntityKind does not contain a GUID,
-     * BadParameter is thrown.
+     * If the given EntityKind does not contain a GUID, BadParameter is thrown.
      *
```
Comment (IkerLuengo) [inline src/cpp/database/database.hpp:280]:
```suggestion
```
Redundant with the @throws specification

File: src/cpp/database/database.hpp [hunk 2]
```diff
@@ -675,17 +719,17 @@ class Database
      * @brief Create the link between a participant and a process. This method is not thread safe.
      *
      * This operation entails:
-     *     1. Adding reference to process to the participant
-     *     2. Adding the participant to the process' list of participants
-     *     3. Adding entry to domains_by_process_
-     *     4. Adding entry to processes_by_domain_
+     *     1. Adding process reference to the participant.
```
Comment (IkerLuengo) [inline src/cpp/database/database.hpp:722]:
```suggestion
     *     1. Referencing the process from the participant.
```

**Review by IkerLuengo — CHANGES_REQUESTED** [review]

**Review by IkerLuengo — APPROVED** [review]
LGTM

Measured coverage loss is due to variability on the tests, not related to the changes.
Also, failing test is known to fail on Windows. A solution is on its way.

**Pull request merged**